pyterrier/rewrite.py

```diff
@@ -250,12 +251,18 @@ def _populate_resultset(self, topics_and_res, qid, index):
             # we need .tolist() as jnius cannot convert numpy arrays
             docids = topics_and_res[topics_and_res["qid"] == qid]["docid"].values.tolist()
             scores = [0.0] * len(docids)
+            if self.requires_scores:
+                scores = topics_and_res[topics_and_res["qid"] == qid]["score"].values.tolist()
             occurrences = [0] * len(docids)
 
         elif "docno" in topics_and_res.columns:
             docnos = topics_and_res[topics_and_res["qid"] == qid]["docno"].values
             docids = []
             scores = []
+            docnos_to_scores = {i:0.0 for i in docnos}
+            if self.requires_scores:
+                docnos_to_scores = {i['docno']: i['score'] for _, i in topics_and_res[topics_and_res["qid"] == qid].iterrows()}
```

Comment:
A bit of an aside, but the `topics_and_res[topics_and_res["qid"] == qid]` selector is also performed a few lines above, and it's pretty inefficient. Can we cache the result to avoid doing it twice?

Reply:
Indeed, we could indeed cache the results.

And I also see now that we do not need a dict, I confused the `skipped` part, I was thinking that the if condition for skipped documents would have a continue statement, but now I see that this is not the case. I will modify the code.

Reply:
I modified the code accordingly.